pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/builder/MultipleTreesBuilder.java

```diff
@@ -83,7 +86,11 @@ public MultipleTreesBuilder(List<StarTreeV2BuilderConfig> builderConfigs, File i
     _segmentDirectory = SegmentDirectoryPaths.findSegmentDirectory(indexDir);
     _metadataProperties =
         CommonsConfigurationUtils.fromFile(new File(_segmentDirectory, V1Constants.MetadataKeys.METADATA_FILE_NAME));
-    Preconditions.checkState(!_metadataProperties.containsKey(MetadataKey.STAR_TREE_COUNT), "Star-tree already exists");
+    _separator = getSeparator();
+    if (_separator != null) {
+      StarTreeBuilderUtils.removeStarTrees(indexDir);
+      _metadataProperties.reload();
```

Comment:
Why do we need to reload the metadata?

Reply:
To ensure that the startree index props removed by `StarTreeBuilderUtils.removeStarTrees` are reflected in the `_metadataProperties`.